mitodl/ol-data-platform
# Create MITx Online user activities intermediate model, updates to xPro

# What are the relevant tickets?




https://github.com/mitodl/hq/issues/1672

# Description (What does it do?)

This PR adds 
- int__mitxonline__user_courseactivities
    - `courseactivity_num_days_activity`
    - `courseactivity_num_events`
    - `courseactivity_num_unique_play_video`
    - `courseactivity_num_play_video`
    - `courseactivity_last_play_video_timestamp`
    - `courseactivity_last_problem_check_timestamp`
    - `courseactivity_first_event_timestamp`
    - `courseactivity_last_event_timestamp`
- int__mitxonline__user_courseactivity_problemcheck
    - user_username
    - courserun_readable_id
    - useractivity_problem_name
    - useractivity_problem_id
    - useractivity_problem_attempts
    - useractivity_problem_success
    - useractivity_problem_current_grade
- int__mitxonline__user_courseactivity_video
    - user_username
    - courserun_readable_id
    - useractivity_video_id
    - useractivity_video_duration
    - useractivity_video_currenttime

# How can this be tested?


To test against production data, you will need to update your [profile.xml](https://github.com/mitodl/ol-data-platform/blob/main/src/ol_dbt/profiles.yml#L46) with production host mitol-ol-data-lake-production.trino.galaxy.starburst.io due to resource limitation of our QA cluster to run large model (https://github.com/mitodl/ol-data-platform/issues/793 is created to solve this problem without updating profiles.yml)
then run
```
 dbt build --select +int__mitxonline__user_courseactivities --vars 'schema_suffix: YOURSCHEMA' --target dev_production 

 dbt build --select +int__mitxpro__user_courseactivities --vars 'schema_suffix: YOURSCHEMA' --target dev_production
```
Tests should all pass
